### PR TITLE
[BugFix] fix memory statistic in local passthrough

### LINKED ISSUE #7181: Fix the memory statistic of LocalPassThrough
### Steps to reproduce the behavior (Required)

Local passthrough mechanism could transfer chunk between fragments without copy memory,  but the memory statistic is not correct. 

The sender fragment allocates memory and record it in its MemTracker, and the receiver fragment release the memory. This could make both MemTracker not correct.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7181

## Problem Summary(Required) ：

Problem:
- Chunks are transferred from one fragment to the other, but not be accounted at `MemTracker`

Solution:
- Release in `MemTracker` at sender fragment
- Allocate in MemTracker in receiver fragment

Considerations:
- The passthrough memory is not accounted at the global `MemTracker` ? 
  - Total memory of passthrough memory is limited by the `max_transmit_batched_bytes`, and it's not a large part
- What if the `PassThroughSenderChannel` destruct without `pull_chunk`, will be lead to memory leak at `MemTracker`?
  - It could not, cause we have accounted the memory at `~PassThroughSenderChannel()`.

### CODE REVIEW

File: be/src/runtime/local_pass_through_buffer.cpp [hunk 1]
```diff
@@ -4,33 +4,50 @@
...
+        // Consume physical bytes in current MemTracker, since later it would be released
+        CurrentThread::current().mem_consume(_physical_bytes);
+        _physical_bytes = 0;
     }
 
 private:
     std::mutex _mutex; // lock-step to push/pull chunks
     ChunkUniquePtrVector _buffer;
     std::vector<size_t> _bytes;
+    int64_t _physical_bytes = 0; // Physical consumed bytes for each chunk
```
Comment (stdpain) [inline be/src/runtime/local_pass_through_buffer.cpp:50]: Is it possiable cause a race condition here?

File: be/src/runtime/current_thread.h [hunk 2]
```diff
@@ -154,10 +158,13 @@ class CurrentThread {
         return res;
     }
 
+    int64_t get_consumed_bytes() const { return _consumed_bytes; }
+
 private:
     const static int64_t BATCH_SIZE = 2 * 1024 * 1024;
 
     int64_t _cache_size = 0;
+    int64_t _consumed_bytes = 0;
```
Comment (kangkaisen) [inline be/src/runtime/current_thread.h:167]: Maybe we need some comments to explain `_consumed_bytes` and `_try_consume_mem_size`.

File: be/src/runtime/local_pass_through_buffer.cpp [hunk 3]
```diff
@@ -4,33 +4,50 @@
...
+    PassThroughSenderChannel() = default;
+
+    ~PassThroughSenderChannel() {
+        if (_physical_bytes > 0) {
+            CurrentThread::current().mem_consume(_physical_bytes);
         }
     }
 
+    void append_chunk(const vectorized::Chunk* chunk, size_t chunk_size, int32_t driver_sequence) {
+        // Release allocated bytes in current MemTracker, since it would not be released at current MemTracker
```
Comment (kangkaisen) [inline be/src/runtime/local_pass_through_buffer.cpp:24]: Is this comment right?

**Review by ZiheLiu — DISMISSED** [review]

**Comment by mofeiatwork** [comment]
run all

File: be/src/runtime/local_pass_through_buffer.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/runtime/local_pass_through_buffer.cpp:50]: All of them are protected with the `_mutex`.

File: be/src/runtime/local_pass_through_buffer.cpp (hunk 3 above)
Comment (mofeiatwork) [reply be/src/runtime/local_pass_through_buffer.cpp:24]: Yes, since we call the `mem_release` later.

**Review by kangkaisen — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.3

**Pull request merged**